permissions_validators/src/private_blockchain/mod.rs

```diff
@@ -38,3 +38,421 @@ impl<W: WorldTrait> IsGrantAllowed<W> for ProhibitGrant {
+        }
+
+        #[test]
+        fn find_account_by_name() {
+            let (alice_id, bob_id, carol_id, wsv, validator) = make_test_env_and_get_3_accounts();
+
+            let op = QueryBox::FindAccountsByName(FindAccountsByName::new(alice_id.clone()));
+
+            assert!(validator.check(&alice_id, &op, &wsv).is_err());
+            assert!(validator.check(&bob_id, &op, &wsv).is_err());
+            assert!(validator.check(&carol_id, &op, &wsv).is_err());
+        }
+
+        #[test]
+        fn find_accounts_by_domain_id() {
+            let (
+                alice_id,
+                bob_id,
+                carol_id,
+                (first_domain_id, _first_domain),
+                (second_domain_id, _second_domain),
+                wsv,
+                validator,
+            ) = make_test_env_and_get_3_accounts_2_domains();
+
+            let op = QueryBox::FindAccountsByDomainId(FindAccountsByDomainId::new(first_domain_id));
+
+            assert!(validator.check(&alice_id, &op, &wsv).is_err());
+            assert!(validator.check(&bob_id, &op, &wsv).is_err());
+            assert!(validator.check(&carol_id, &op, &wsv).is_err());
+
+            let op =
+                QueryBox::FindAccountsByDomainId(FindAccountsByDomainId::new(second_domain_id));
+
+            assert!(validator.check(&alice_id, &op, &wsv).is_err());
+            assert!(validator.check(&bob_id, &op, &wsv).is_err());
+            assert!(validator.check(&carol_id, &op, &wsv).is_err());
+        }
+
+        #[test]
+        fn find_accounts_with_asset() {
+            let (alice_id, bob_id, carol_id, wsv, validator) = make_test_env_and_get_3_accounts();
+
+            let op = QueryBox::FindAccountsWithAsset(FindAccountsWithAsset::new("xor".to_owned()));
+
+            assert!(validator.check(&alice_id, &op, &wsv).is_err());
+            assert!(validator.check(&bob_id, &op, &wsv).is_err());
+            assert!(validator.check(&carol_id, &op, &wsv).is_err());
+        }
+    }
+
+    #[test]
+    fn probihit_register_domains_failed() {
+        let alice_id = AccountId::from_str("alice@test0").expect("Valid");
+
+        let instruction = Instruction::Register(RegisterBox::new(Domain::new(
+            "new_domain".parse().expect("Valid"),
+        )));
+
+        let wsv = WorldStateView::<World>::new(World::new());
+
+        assert!(register::ProhibitRegisterDomains
+            .check(&alice_id, &instruction, &wsv)
+            .is_err());
+    }
+
+    #[test]
+    fn prohibit_register_domains_success() {
+        let alice_id = AccountId::from_str("alice@test0").expect("Valid");
+
+        let instruction = Instruction::Register(RegisterBox::new(Account::new(
+            "bob@test".parse().expect("Valid"),
+            [],
+        )));
+
+        let wsv = WorldStateView::<World>::new(World::new());
+
+        assert!(register::ProhibitRegisterDomains
+            .check(&alice_id, &instruction, &wsv)
+            .is_err());
```

Comment:
By the way, I don't understand why this is forbidden to register account with `ProhibitRegisterDomains` permission. Looks like a bug.

Reply:
Maybe a good idea to ask in the chat.